Mathlib/Lean/Meta/CongrTheorems.lean

```diff
@@ -69,4 +71,192 @@ where
     -- We have no more tricks.
     failure
 
+/--
+Create a congruence lemma to prove that `Eq/HEq (f a₁ ... aₙ) (f' a₁' ... aₙ')`.
```

Comment:
For a long doc-string, I like to start with something like "`mkRichHCongr fType info fixedFun fixedParams forceHEq` creates ...", so the name of the function and the arguments are immediately in the reader's input context.